src/app/stripe.service.ts

```diff
@@ -86,71 +81,6 @@ export class StripeService {
     this.lastCardCountry = cardCountry;
   }
 
-  async confirmPaymentWithNewCardOrPRB(
-    donationPreUpdate: Donation,
-    cardElement: StripeCardElement,
-  ): Promise<PaymentIntentResult | undefined> {
-    if (!this.stripe || !donationPreUpdate.clientSecret || !donationPreUpdate.donationId) {
-      console.log('Stripe not ready for confirmPayment()');
-      return;
-    }
-
-    let paymentMethod: any;
-    let isPrb = false;
-
-    const billingDetails: PaymentMethodCreateParams.BillingDetails = {
-      email: donationPreUpdate.emailAddress,
-      name: `${donationPreUpdate.firstName} ${donationPreUpdate.lastName}` ?? undefined,
-    };
-
-    billingDetails.address = {
-      country: donationPreUpdate.countryCode,
-      postal_code: donationPreUpdate.billingPostalAddress, // Just postcode in the Stripe case.
-    };
-
-    // Ensure fee info is updated before finalising payment.
-    return new Promise<PaymentIntentResult>((resolve, reject) => {
-      this.donationService.updatePaymentDetails(donationPreUpdate, this.lastCardBrand, this.lastCardCountry)
-      .subscribe(donation => {
-        if (!donation.clientSecret || !donation.donationId) {
-          reject('Missing ID in card-details-updated donation');
-          return;
-        }
-
-        // Processing adapted from
-        // https://stripe.com/docs/stripe-js/elements/payment-request-button?html-or-react=html#html-js-complete-payment
-        // and card version merged with PRB one. More detailed comments on params
-        // rationale there.
-        if (this.paymentMethodIds.has(donation.donationId)) {
-          isPrb = true;
-          paymentMethod = this.paymentMethodIds.get(donation.donationId);
-        } else {
-          paymentMethod = {
-            card: cardElement,
-            // See https://stripe.com/docs/payments/accept-a-payment#web-submit-payment
-            billing_details: billingDetails,
-          };
-        }
-
-        let thePaymentMethodEvent = this.paymentMethodEvents.get(donation.donationId);
```

Comment:
I think `paymentMethodEvents` is redundant now too.

Probably `paymentMethodIds` and the code managing them as well?

Reply:
You're right: 79f3c405